native/cocos/base/LogRemote.cpp

```diff
@@ -0,0 +1,296 @@
+            } else {
+                _clientID = "flagId is not set!";
+            }
+
+            if (doc.HasMember(ATC_KEY_PLANID)) {
+                _testID = doc[ATC_KEY_PLANID].GetString();
+            } else {
+                _testID = "planId is not set!";
+            }
+        }
+
+        {
+            // parse ip & port
+            rapidjson::Value &cfg = doc[ATC_KEY_CONFIG];
+            if (!cfg.HasMember(ATC_KEY_IP) || !cfg.HasMember(ATC_KEY_PORT)) {
+                _status = UdpLogClientState::DONE;
+                return;
+            }
+            const char *remoteIp = cfg[ATC_KEY_IP].GetString();
+            int remotePort = cfg[ATC_KEY_PORT].GetInt() + 1;
+            setServerAddr(remoteIp, remotePort);
+        }
+        _bootID = std::chrono::duration_cast<std::chrono::milliseconds>(
+                      std::chrono::system_clock::now().time_since_epoch())
+                      .count();
+
+        _status = UdpLogClientState::CONFIGURED;
+    }
+
+    void setServerAddr(const std::string_view &addr, int port) {
+        memset(&_serverAddr, 0, sizeof(_serverAddr));
+        _serverAddr.sin_family = AF_INET;
+        _serverAddr.sin_addr.s_addr = inet_addr(addr.data());
+        _serverAddr.sin_port = htons(port);
+    }
+
+    static auto getErrorCode() {
+    #if CC_PLATFORM == CC_PLATFORM_WINDOWS
+        return WSAGetLastError();
+    #else
+        return errno;
+    #endif
+    }
+
+    void ensureInitSocket() {
+        if (_status == UdpLogClientState::INITIALIZED) {
+            tryParseConfig();
+        }
+        if (_status != UdpLogClientState::CONFIGURED) {
+            return;
+        }
+        memset(&_localAddr, 0, sizeof(_localAddr));
+        _localAddr.sin_family = AF_INET;
+        _localAddr.sin_addr.s_addr = INADDR_ANY;
+        _localAddr.sin_port = 0; // bind random port
+
+        if ((_sock = socket(AF_INET, SOCK_DGRAM, 0)) < 0) {
+            auto errorCode = getErrorCode();
+            _status = UdpLogClientState::DONE;
+            printf("create socket failed, code: %d\n", errorCode);
+            return;
+        }
+        if (bind(_sock, reinterpret_cast<sockaddr *>(&_localAddr), sizeof(_localAddr))) {
+            _status = UdpLogClientState::DONE;
+            auto errorCode = getErrorCode();
+            printf("bind socket failed, code: %d\n", errorCode);
+            return;
+        }
+        if (connect(_sock, reinterpret_cast<sockaddr *>(&_serverAddr), sizeof(_serverAddr))) {
+            auto errorCode = getErrorCode();
+            printf("connect socket failed, code: %d\n", errorCode);
+            _status = UdpLogClientState::DONE;
+            return;
+        }
+
+    #if CC_PLATFORM == CC_PLATFORM_WINDOWS
+        u_long mode = 1;
+        if (ioctlsocket(_sock, FIONBIO, &mode)) {
+            auto errorCode = getErrorCode();
+            printf("set nonblock failed, code: %d\n", errorCode);
+            // continue
```

Comment:
If ioctlsocket fails, this function still set `_status` to UdpLogClientState::OK , and run `send(_sock, msg.data(), msg.size(), 0);` logic ?

Reply:
Setting the fd to nonblock is not required